diff --git a/bashrc b/bashrc
index d19a7c1..d462ac7 100644
--- a/bashrc
+++ b/bashrc
@@ -19,8 +19,6 @@ if [ -z "$BASHRCSOURCED" ]; then
       xterm*|vte*)
         if [ -e /etc/sysconfig/bash-prompt-xterm ]; then
             PROMPT_COMMAND=/etc/sysconfig/bash-prompt-xterm
-        elif [ "${VTE_VERSION:-0}" -ge 3405 ]; then
-            PROMPT_COMMAND="__vte_prompt_command"
         else
             PROMPT_COMMAND='printf "\033]0;%s@%s:%s\007" "${USER}" "${HOSTNAME%%.*}" "${PWD/#$HOME/\~}"'
         fi
